src/libfuncs/felt252.rs

```diff
@@ -307,43 +274,25 @@ pub fn build_binary_operation<'ctx, 'this>(
             // egcd sometimes returns a negative number for the inverse,
             // in such cases we must simply wrap it around back into [0, PRIME)
             // this suffices because |inv_i| <= divfloor(PRIME,2)
-            let zero = negative_check_block
-                .append_operation(arith::constant(
-                    context,
-                    IntegerAttribute::new(i512, 0).into(),
-                    location,
-                ))
-                .result(0)?
-                .into();
-
-            let is_negative = negative_check_block
-                .append_operation(arith::cmpi(
-                    context,
-                    CmpiPredicate::Slt,
-                    inverse,
-                    zero,
-                    location,
-                ))
-                .result(0)?
-                .into();
+            let zero = negative_check_block.const_int(context, location, 0, 512)?;
+
+            let is_negative = negative_check_block.append_op_result(arith::cmpi(
+                context,
+                CmpiPredicate::Slt,
+                inverse,
+                zero,
+                location,
+            ))?;
             // if the inverse is < 0, add PRIME
-            let prime = negative_check_block
-                .append_operation(arith::constant(context, attr_prime_i512, location))
-                .result(0)?
-                .into();
-            let wrapped_inverse = negative_check_block
-                .append_operation(arith::addi(inverse, prime, location))
-                .result(0)?
-                .into();
-            let inverse = negative_check_block
-                .append_operation(arith::select(
-                    is_negative,
-                    wrapped_inverse,
-                    inverse,
-                    location,
-                ))
-                .result(0)?
-                .into();
+            let prime = negative_check_block.const_int(context, location, 0, 512)?;
```

Comment:
Here in fact, even better you can use `const_int_from_type`

Reply:
It says in the block_ext file that it should not be used for felt252, should I still use it?